flags/state.py

```diff
@@ -3,8 +3,16 @@
 
 def flag_state(flag_name, **kwargs):
     """ Return the value for the flag by passing kwargs to its conditions """
+
+    # If the request is given as a kwargs, and the FlagConditionsMiddleware is
+    # enabled, use the flag conditions attached to the request.
+    if hasattr(kwargs.get('request'), 'FLAG_CONDITIONS'):
+        conditions = kwargs['request'].FLAG_CONDITIONS
+    else:
+        conditions = get_flags()
+
     try:
-        return get_flags()[flag_name].check_state(**kwargs)
+        return conditions[flag_name].check_state(**kwargs)
```

Comment:
I noticed when reviewing this that this logic would improperly catch `KeyError`s raised by `check_state`, when really we only want to catch the `KeyError` if we look up a non-existent key. If for whatever reason a condition (most likely erroneously) raises `KeyError`, we'd want it to pass through to the caller and not be improperly treated as a false flag. See [this](https://github.com/cfpb/django-flags/compare/flags-middleware...chosak:flags-middleware-review#diff-592e4b7a150c71c1b47086b383f2a9b5R23) for a suggested implementation.